sql/item.cc

```diff
@@ -2176,6 +2176,28 @@ Item::Type Item_name_const::type() const
 
 bool Item_name_const::fix_fields(THD *thd, Item **ref)
 {
+  /*
+    If we have either of the following:
+      ... WHERE foo=NAME_CONST(...)
+      ... JOIN ... ON foo=NAME_CONST(...)
+    then we have an opportunity to unwrap the NAME_CONST and
+    use the enclosed value directly, replacing NAME_CONST in
+    the parse tree with the value it encloses.
+   */
+  if ((thd->where == THD_WHERE::WHERE_CLAUSE ||
+       thd->where == THD_WHERE::ON_CLAUSE) &&
+      (value_item->type() == FUNC_ITEM ||
+       value_item->type() == CONST_ITEM) &&
+    !value_item->fix_fields_if_needed(thd, &value_item))
```

Comment:
We have a call to `value_item->fix_fields_if_needed`  here, and then again we have it below...  I think having all fix_fields calls first and rewrite second would be cleaner...

Reply:
Yes I agree, this is nicer, thank you.